feat: multiply conv reward with priority for higher reward

ishowvel opened the pull request:
Resolves #175

ishowvel commented:
/help

ishowvel commented:
isnt there a /ask command? i remember there being one

ishowvel commented:
heres the [qa](https://github.com/ishowvel-org/.ubiquity-os/issues/10)

ishowvel commented:
@gentlementlegen please review this :pray:

gentlementlegen commented on src/helpers/label-price-extractor.ts:
```diff
@@ -20,3 +20,47 @@ export function getSortedPrices(labels: GitHubIssue["labels"] | undefined) {
 }
+
+export function parsePriorityLabel(
+  labels:
```
Please use `octokit` types.

gentlementlegen commented:
@ishowvel You should add `Jest` tests to your pull-request.

Here is my latest QA:
![image](https://github.com/user-attachments/assets/5fff37e0-5764-4e8e-8e2f-73a6d9cdf39a)

The decimals should not look like this. Also, I am confused about the total here. Can you walk me through the calculations?

ishowvel commented:
After the total formatted reward is calculated, the priority level is multiplied to the reward for that specific comment, this all happens in the formatting evaluation.
Then after the reward is passed to content evaluation to calc and multiply it with the relevance for that specific comment.

The only change here is adding a multiplication of the priority level to the formatting reward calculation

ishowvel commented:
the total itself is not multiplied but the specification and comment incentives are

ishowvel commented:
for jest testing, no mock result has ever contained the "priority" key. Can you help me find a way to make the mock results for this?
all results starting at the formatting module would need to be changed

0x4007 commented:
Just copy the priority level names? Our config is public you can see all of them .ubiquity-os repo

ishowvel commented:
https://github.com/ubiquity-os-marketplace/text-conversation-rewards/blob/development/tests%2F__mocks__%2Fresults%2Fformatting-evaluator-results.json

0x4007 commented:
There's five priority levels

ishowvel commented:
I am talking about the results, each result object would need to be edited to have an extra priority key and the total and reward values would need to be edited

ishowvel commented:
<details><summary><b><h3>&nbsp;<a href=\"https://pay.ubq.fi?claim=W3sidHlwZSI6ImVyYzIwLXBlcm1pdCIsInBlcm1pdCI6eyJwZXJtaXR0ZWQiOnsidG9rZW4iOiIweGU5MUQxNTNFMGI0MTUxOEEyQ2U4RGQzRDc5NDRGYTg2MzQ2M2E5N2QiLCJhbW91bnQiOiI4MTg0ODIwMDAwMDAwMDAwMDAwMDAifSwibm9uY2UiOiI4MzA3NjQzNzQ0NjQ5OTU4OTMwNDYxMTEyODk2MzkxNjcxMDEwNjg4NjEwMjQzNjAyMzg4MTUyMDA1ODM0NjgwMDU3ODc1NDQwMTc1NSIsImRlYWRsaW5lIjoiMTE1NzkyMDg5MjM3MzE2MTk1NDIzNTcwOTg1MDA4Njg3OTA3ODUzMjY5OTg0NjY1NjQwNTY0MDM5NDU3NTg0MDA3OTEzMTI5NjM5OTM1In0sInRyYW5zZmVyRGV0YWlscyI6eyJ0byI6IjB4NEQwNzA0ZjQwMEQ1N0JhOTNlRWE4ODc2NUMzRmNEQkQ4MjZkQ0ZjNCIsInJlcXVlc3RlZEFtb3VudCI6IjgxODQ4MjAwMDAwMDAwMDAwMDAwMCJ9LCJvd25lciI6IjB4ZDk1MzBGM2ZiQkVhMTFiZUQwMURDMDlFNzkzMThmMmYyMDIyMzcxNiIsInNpZ25hdHVyZSI6IjB4NjczNTJkNzZjYzM5Yjg1MWI1NzZlNGYyZjg2M2Y2ZjcyNGNjOWFhNGIwYTgyOTNhNjE3NDcwMTEzYjZkNmJjNTVmNTYyY2FiZTU4OGIxZTdjZDg4YjJmNzQ5ZDlmODcwMTQ1YjdiNzBkMGU3OWY5MDFkOTU3MTYzOGEyMzVkZmIxYiIsIm5ldHdvcmtJZCI6MTAwfV0=\" target=\"_blank\" rel=\"noopener\">[ 818.482 WXDAI ]</a>&nbsp;</h3><h6>@gentlementlegen</h6></b></summary><h6>Contributions Overview</h6><table><thead><tr><th>View</th><th>Contribution</th><th>Count</th><th>Reward</th></tr></thead><tbody><tr><td>Issue</td><td>Task</td><td>1</td><td>400</td></tr><tr><td>Issue</td><td>Specification</td><td>1</td><td>15.93</td></tr><tr><td>Issue</td><td>Comment</td><td>9</td><td>112.864</td></tr><tr><td>Review</td><td>Comment</td><td>25</td><td>289.688</td></tr></tbody></table><h6>Conversation Incentives</h6><table><thead><tr><th>Comment</th><th>Formatting</th><th>Relevance</th><th>Reward</th></tr></thead><tbody><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/issues/5\" target=\"_blank\" rel=\"noopener\">In the v1 of the Ubiquibot, when a result gets evaluated, a reca&hellip;</a></h6></td><td><details><summary>15.93</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 6&#13;    a:&#13;      score: 1&#13;      elementCount: 1&#13;    ul:&#13;      score: 1&#13;      elementCount: 1&#13;    li:&#13;      score: 1&#13;      elementCount: 3&#13;  result: 11&#13;regex:&#13;  wordCount: 98&#13;  wordValue: 0.1&#13;  result: 4.93&#13;</pre></details></td><td>1</td><td>15.93</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/issues/5#issuecomment-2033404518\" target=\"_blank\" rel=\"noopener\">This needs https://github.com/ubiquity-os/conversation-rewards/p&hellip;</a></h6></td><td><details><summary>5.11</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 1&#13;regex:&#13;  wordCount: 35&#13;  wordValue: 0.2&#13;  result: 4.11&#13;</pre></details></td><td>0.8</td><td>4.288</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/issues/5#issuecomment-2036174312\" target=\"_blank\" rel=\"noopener\">To me 1 is the most straightforward to do for few reasons:- th&hellip;</a></h6></td><td><details><summary>19.95</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 5&#13;    ul:&#13;      score: 1&#13;      elementCount: 1&#13;    li:&#13;      score: 1&#13;      elementCount: 3&#13;  result: 9&#13;regex:&#13;  wordCount: 111&#13;  wordValue: 0.2&#13;  result: 10.95&#13;</pre></details></td><td>0.8</td><td>17.76</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/issues/5#issuecomment-2036367126\" target=\"_blank\" rel=\"noopener\">I think each plugin should output JSON not html as it is not rel&hellip;</a></h6></td><td><details><summary>17.29</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 3&#13;    code:&#13;      score: 1&#13;      elementCount: 3&#13;  result: 6&#13;regex:&#13;  wordCount: 115&#13;  wordValue: 0.2&#13;  result: 11.29&#13;</pre></details></td><td>0.8</td><td>15.032</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/issues/5#issuecomment-2036385985\" target=\"_blank\" rel=\"noopener\">If you want to manipulate and convey data, HTML really is not ma&hellip;</a></h6></td><td><details><summary>17.85</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 4&#13;    a:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 5&#13;regex:&#13;  wordCount: 134&#13;  wordValue: 0.2&#13;  result: 12.85&#13;</pre></details></td><td>0.8</td><td>15.28</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/issues/5#issuecomment-2036411811\" target=\"_blank\" rel=\"noopener\">But then how do we consider the formatting of that output?Prac&hellip;</a></h6></td><td><details><summary>19.87</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 5&#13;  result: 5&#13;regex:&#13;  wordCount: 159&#13;  wordValue: 0.2&#13;  result: 14.87&#13;</pre></details></td><td>0.8</td><td>16.896</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/issues/5#issuecomment-2036458775\" target=\"_blank\" rel=\"noopener\">This can work, but we skyrocket coupling and to me defeat purpos&hellip;</a></h6></td><td><details><summary>7.03</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 1&#13;regex:&#13;  wordCount: 55&#13;  wordValue: 0.2&#13;  result: 6.03&#13;</pre></details></td><td>0.8</td><td>5.824</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/issues/5#issuecomment-2036535332\" target=\"_blank\" rel=\"noopener\">@pavlovcik To mitigate that that's why inside the comment reward&hellip;</a></h6></td><td><details><summary>10.73</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;    code:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 2&#13;regex:&#13;  wordCount: 85&#13;  wordValue: 0.2&#13;  result: 8.73&#13;</pre></details></td><td>0.8</td><td>8.984</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/issues/5#issuecomment-2051094255\" target=\"_blank\" rel=\"noopener\">I realized that to carry this task properly we need to handle fl&hellip;</a></h6></td><td><details><summary>9.31</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;    code:&#13;      score: 1&#13;      elementCount: 2&#13;  result: 3&#13;regex:&#13;  wordCount: 58&#13;  wordValue: 0.2&#13;  result: 6.31&#13;</pre></details></td><td>0.8</td><td>8.048</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/issues/5#issuecomment-2054424028\" target=\"_blank\" rel=\"noopener\">Agreed, I think currently there are 3 possible things to annotat&hellip;</a></h6></td><td><details><summary>21.94</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 5&#13;    ul:&#13;      score: 1&#13;      elementCount: 1&#13;    li:&#13;      score: 1&#13;      elementCount: 3&#13;    code:&#13;      score: 1&#13;      elementCount: 7&#13;  result: 16&#13;regex:&#13;  wordCount: 54&#13;  wordValue: 0.2&#13;  result: 5.94&#13;</pre></details></td><td>0.8</td><td>20.752</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12\" target=\"_blank\" rel=\"noopener\">Resolves #5 Depends on #8 Reviews looks gigantic but it's on&hellip;</a></h6></td><td><details><summary>3</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 3&#13;  result: 3&#13;regex:&#13;  wordCount: 19&#13;  wordValue: 0&#13;  result: 0&#13;</pre></details></td><td>0.7</td><td>0</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1570010044\" target=\"_blank\" rel=\"noopener\">I think in this case it makes sense because this function will r&hellip;</a></h6></td><td><details><summary>12.49</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 2&#13;    code:&#13;      score: 1&#13;      elementCount: 4&#13;  result: 6&#13;regex:&#13;  wordCount: 60&#13;  wordValue: 0.2&#13;  result: 6.49&#13;</pre></details></td><td>0.7</td><td>21.086</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1570011467\" target=\"_blank\" rel=\"noopener\">It is supposed to represent a comment. Would you prefer a descri&hellip;</a></h6></td><td><details><summary>4.11</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;    code:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 2&#13;regex:&#13;  wordCount: 16&#13;  wordValue: 0.2&#13;  result: 2.11&#13;</pre></details></td><td>0.7</td><td>6.954</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1570014142\" target=\"_blank\" rel=\"noopener\">True, just picked this up from the previous codebase. https://rp&hellip;</a></h6></td><td><details><summary>3.44</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 1&#13;regex:&#13;  wordCount: 19&#13;  wordValue: 0.2&#13;  result: 2.44&#13;</pre></details></td><td>0.7</td><td>5.416</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1570187141\" target=\"_blank\" rel=\"noopener\">Seems convoluted to me and I don't know what problem that solved&hellip;</a></h6></td><td><details><summary>7.6</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 2&#13;    code:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 3&#13;regex:&#13;  wordCount: 40&#13;  wordValue: 0.2&#13;  result: 4.6&#13;</pre></details></td><td>0.7</td><td>12.44</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1570188450\" target=\"_blank\" rel=\"noopener\">Changed &#96;COMMENT&#96; to &#96;COMMENTED&#96;, added jsdoc de&hellip;</a></h6></td><td><details><summary>6.98</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 2&#13;    code:&#13;      score: 1&#13;      elementCount: 2&#13;  result: 4&#13;regex:&#13;  wordCount: 24&#13;  wordValue: 0.2&#13;  result: 2.98&#13;</pre></details></td><td>0.7</td><td>12.172</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1573413974\" target=\"_blank\" rel=\"noopener\">Simply makes the docs linking the code within the comment, for c&hellip;</a></h6></td><td><details><summary>8.37</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 3&#13;  result: 3&#13;regex:&#13;  wordCount: 48&#13;  wordValue: 0.2&#13;  result: 5.37&#13;</pre></details></td><td>0.7</td><td>13.518</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1573416633\" target=\"_blank\" rel=\"noopener\">In the case of a string it would coerce the value and you would &hellip;</a></h6></td><td><details><summary>4.09</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 1&#13;regex:&#13;  wordCount: 25&#13;  wordValue: 0.2&#13;  result: 3.09&#13;</pre></details></td><td>0.7</td><td>6.326</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1573420718\" target=\"_blank\" rel=\"noopener\">&#96;observe&#96; would help to chose what we want to include an&hellip;</a></h6></td><td><details><summary>7.7</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;    code:&#13;      score: 1&#13;      elementCount: 1&#13;    a:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 3&#13;regex:&#13;  wordCount: 41&#13;  wordValue: 0.2&#13;  result: 4.7&#13;</pre></details></td><td>0.7</td><td>12.58</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1573731420\" target=\"_blank\" rel=\"noopener\">Not very useful indeed. I am working on the tests on a separate &hellip;</a></h6></td><td><details><summary>6.31</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 2&#13;  result: 2&#13;regex:&#13;  wordCount: 37&#13;  wordValue: 0.2&#13;  result: 4.31&#13;</pre></details></td><td>0.7</td><td>10.034</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1573731649\" target=\"_blank\" rel=\"noopener\">See https://github.com/ubiquity-os/conversation-rewards/pull/14 &hellip;</a></h6></td><td><details><summary>3.22</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 1&#13;regex:&#13;  wordCount: 17&#13;  wordValue: 0.2&#13;  result: 2.22&#13;</pre></details></td><td>0.7</td><td>5.108</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1573733603\" target=\"_blank\" rel=\"noopener\">Good point, also symbol should not be DAI but ETH isn't it?&#9&hellip;</a></h6></td><td><details><summary>7.89</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 2&#13;    code:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 3&#13;regex:&#13;  wordCount: 43&#13;  wordValue: 0.2&#13;  result: 4.89&#13;</pre></details></td><td>0.7</td><td>12.846</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1573737049\" target=\"_blank\" rel=\"noopener\">Had changed it everywhere else but here, now it should be fine</a></h6></td><td><details><summary>2.65</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 1&#13;regex:&#13;  wordCount: 12&#13;  wordValue: 0.2&#13;  result: 1.65&#13;</pre></details></td><td>0.7</td><td>4.31</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1574114225\" target=\"_blank\" rel=\"noopener\">This matches any whitespace character (spaces, tabs, line breaks&hellip;</a></h6></td><td><details><summary>4.91</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 1&#13;regex:&#13;  wordCount: 33&#13;  wordValue: 0.2&#13;  result: 3.91&#13;</pre></details></td><td>0.7</td><td>7.474</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1574622869\" target=\"_blank\" rel=\"noopener\">By definition &#96;dump&#96; was just outputting results to stdo&hellip;</a></h6></td><td><details><summary>9.22</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;    code:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 2&#13;regex:&#13;  wordCount: 68&#13;  wordValue: 0.2&#13;  result: 7.22&#13;</pre></details></td><td>0.7</td><td>14.108</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1574702577\" target=\"_blank\" rel=\"noopener\">Un-nested them one level. I think that the &#96;buildXrow&#96; s&hellip;</a></h6></td><td><details><summary>5.09</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;    code:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 2&#13;regex:&#13;  wordCount: 25&#13;  wordValue: 0.2&#13;  result: 3.09&#13;</pre></details></td><td>0.7</td><td>8.326</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1574724902\" target=\"_blank\" rel=\"noopener\">I assumed BigInt only handles integers, and we manipulate a lot &hellip;</a></h6></td><td><details><summary>3.98</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 1&#13;regex:&#13;  wordCount: 24&#13;  wordValue: 0.2&#13;  result: 2.98&#13;</pre></details></td><td>0.7</td><td>6.172</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1574767984\" target=\"_blank\" rel=\"noopener\">There is one permit generated per participating user, last run:&hellip;</a></h6></td><td><details><summary>4.44</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 2&#13;  result: 2&#13;regex:&#13;  wordCount: 19&#13;  wordValue: 0.2&#13;  result: 2.44&#13;</pre></details></td><td>0.7</td><td>7.416</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1575659438\" target=\"_blank\" rel=\"noopener\">I made it according to the previous version:- Issue Specificat&hellip;</a></h6></td><td><details><summary>12.8</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 5&#13;    ul:&#13;      score: 1&#13;      elementCount: 1&#13;    li:&#13;      score: 1&#13;      elementCount: 2&#13;  result: 8&#13;regex:&#13;  wordCount: 42&#13;  wordValue: 0.2&#13;  result: 4.8&#13;</pre></details></td><td>0.7</td><td>22.72</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1575675704\" target=\"_blank\" rel=\"noopener\">Wouldn't that lead to precision loss in JS and after the divisio&hellip;</a></h6></td><td><details><summary>5.6</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;    code:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 2&#13;regex:&#13;  wordCount: 30&#13;  wordValue: 0.2&#13;  result: 3.6&#13;</pre></details></td><td>0.7</td><td>9.04</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1576567583\" target=\"_blank\" rel=\"noopener\">That is correct. I understand now why you're confused:The valu&hellip;</a></h6></td><td><details><summary>21.95</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 8&#13;    code:&#13;      score: 1&#13;      elementCount: 7&#13;  result: 15&#13;regex:&#13;  wordCount: 65&#13;  wordValue: 0.2&#13;  result: 6.95&#13;</pre></details></td><td>0.7</td><td>39.73</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1577477616\" target=\"_blank\" rel=\"noopener\">I don't think is it relevant in this scenario because we are not&hellip;</a></h6></td><td><details><summary>6.37</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 1&#13;regex:&#13;  wordCount: 48&#13;  wordValue: 0.2&#13;  result: 5.37&#13;</pre></details></td><td>0.7</td><td>9.518</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1578162024\" target=\"_blank\" rel=\"noopener\">@whilefoo you are correct, these were wrongly used. I corrected &hellip;</a></h6></td><td><details><summary>16.31</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 5&#13;    code:&#13;      score: 1&#13;      elementCount: 5&#13;  result: 10&#13;regex:&#13;  wordCount: 58&#13;  wordValue: 0.2&#13;  result: 6.31&#13;</pre></details></td><td>0.7</td><td>28.834</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#issuecomment-2056635516\" target=\"_blank\" rel=\"noopener\">Example of successful comment posting with results:https://git&hellip;</a></h6></td><td><details><summary>4.11</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 2&#13;  result: 2&#13;regex:&#13;  wordCount: 16&#13;  wordValue: 0.2&#13;  result: 2.11&#13;</pre></details></td><td>0.7</td><td>6.954</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#issuecomment-2063348339\" target=\"_blank\" rel=\"noopener\">@0x4007 I believe that if I had all the tests in this PR it will&hellip;</a></h6></td><td><details><summary>4.29</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 1&#13;regex:&#13;  wordCount: 27&#13;  wordValue: 0.2&#13;  result: 3.29&#13;</pre></details></td><td>0.7</td><td>6.606</td></tr></tbody></table></details>

<details><summary><b><h3>&nbsp;<a href=\"undefined\" target=\"_blank\" rel=\"noopener\">[ 2073.928 WXDAI ]</a>&nbsp;</h3><h6>@gentlementlegen</h6></b></summary><h6>Contributions Overview</h6><table><thead><tr><th>View</th><th>Contribution</th><th>Count</th><th>Reward</th></tr></thead><tbody><tr><td>Issue</td><td>Task</td><td>1</td><td>400</td></tr><tr><td>Issue</td><td>Specification</td><td>1</td><td>63.72</td></tr><tr><td>Issue</td><td>Comment</td><td>9</td><td>451.456</td></tr><tr><td>Review</td><td>Comment</td><td>25</td><td>1158.752</td></tr></tbody></table><h6>Conversation Incentives</h6><table><thead><tr><th>Comment</th><th>Formatting</th><th>Relevance</th><th>Priority</th><th>Reward</th></tr></thead><tbody><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/issues/5\" target=\"_blank\" rel=\"noopener\">In the v1 of the Ubiquibot, when a result gets evaluated, a reca&hellip;</a></h6></td><td><details><summary>15.93</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 6&#13;    a:&#13;      score: 1&#13;      elementCount: 1&#13;    ul:&#13;      score: 1&#13;      elementCount: 1&#13;    li:&#13;      score: 1&#13;      elementCount: 3&#13;  result: 11&#13;regex:&#13;  wordCount: 98&#13;  wordValue: 0.1&#13;  result: 4.93&#13;</pre></details></td><td>1</td><td>4</td><td>63.72</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/issues/5#issuecomment-2033404518\" target=\"_blank\" rel=\"noopener\">This needs https://github.com/ubiquity-os/conversation-rewards/p&hellip;</a></h6></td><td><details><summary>5.11</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 1&#13;regex:&#13;  wordCount: 35&#13;  wordValue: 0.2&#13;  result: 4.11&#13;</pre></details></td><td>0.8</td><td>4</td><td>17.152</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/issues/5#issuecomment-2036174312\" target=\"_blank\" rel=\"noopener\">To me 1 is the most straightforward to do for few reasons:- th&hellip;</a></h6></td><td><details><summary>19.95</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 5&#13;    ul:&#13;      score: 1&#13;      elementCount: 1&#13;    li:&#13;      score: 1&#13;      elementCount: 3&#13;  result: 9&#13;regex:&#13;  wordCount: 111&#13;  wordValue: 0.2&#13;  result: 10.95&#13;</pre></details></td><td>0.8</td><td>4</td><td>71.04</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/issues/5#issuecomment-2036367126\" target=\"_blank\" rel=\"noopener\">I think each plugin should output JSON not html as it is not rel&hellip;</a></h6></td><td><details><summary>17.29</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 3&#13;    code:&#13;      score: 1&#13;      elementCount: 3&#13;  result: 6&#13;regex:&#13;  wordCount: 115&#13;  wordValue: 0.2&#13;  result: 11.29&#13;</pre></details></td><td>0.8</td><td>4</td><td>60.128</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/issues/5#issuecomment-2036385985\" target=\"_blank\" rel=\"noopener\">If you want to manipulate and convey data, HTML really is not ma&hellip;</a></h6></td><td><details><summary>17.85</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 4&#13;    a:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 5&#13;regex:&#13;  wordCount: 134&#13;  wordValue: 0.2&#13;  result: 12.85&#13;</pre></details></td><td>0.8</td><td>4</td><td>61.12</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/issues/5#issuecomment-2036411811\" target=\"_blank\" rel=\"noopener\">But then how do we consider the formatting of that output?Prac&hellip;</a></h6></td><td><details><summary>19.87</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 5&#13;  result: 5&#13;regex:&#13;  wordCount: 159&#13;  wordValue: 0.2&#13;  result: 14.87&#13;</pre></details></td><td>0.8</td><td>4</td><td>67.584</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/issues/5#issuecomment-2036458775\" target=\"_blank\" rel=\"noopener\">This can work, but we skyrocket coupling and to me defeat purpos&hellip;</a></h6></td><td><details><summary>7.03</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 1&#13;regex:&#13;  wordCount: 55&#13;  wordValue: 0.2&#13;  result: 6.03&#13;</pre></details></td><td>0.8</td><td>4</td><td>23.296</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/issues/5#issuecomment-2036535332\" target=\"_blank\" rel=\"noopener\">@pavlovcik To mitigate that that's why inside the comment reward&hellip;</a></h6></td><td><details><summary>10.73</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;    code:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 2&#13;regex:&#13;  wordCount: 85&#13;  wordValue: 0.2&#13;  result: 8.73&#13;</pre></details></td><td>0.8</td><td>4</td><td>35.936</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/issues/5#issuecomment-2051094255\" target=\"_blank\" rel=\"noopener\">I realized that to carry this task properly we need to handle fl&hellip;</a></h6></td><td><details><summary>9.31</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;    code:&#13;      score: 1&#13;      elementCount: 2&#13;  result: 3&#13;regex:&#13;  wordCount: 58&#13;  wordValue: 0.2&#13;  result: 6.31&#13;</pre></details></td><td>0.8</td><td>4</td><td>32.192</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/issues/5#issuecomment-2054424028\" target=\"_blank\" rel=\"noopener\">Agreed, I think currently there are 3 possible things to annotat&hellip;</a></h6></td><td><details><summary>21.94</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 5&#13;    ul:&#13;      score: 1&#13;      elementCount: 1&#13;    li:&#13;      score: 1&#13;      elementCount: 3&#13;    code:&#13;      score: 1&#13;      elementCount: 7&#13;  result: 16&#13;regex:&#13;  wordCount: 54&#13;  wordValue: 0.2&#13;  result: 5.94&#13;</pre></details></td><td>0.8</td><td>4</td><td>83.008</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12\" target=\"_blank\" rel=\"noopener\">Resolves #5 Depends on #8 Reviews looks gigantic but it's on&hellip;</a></h6></td><td><details><summary>3</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 3&#13;  result: 3&#13;regex:&#13;  wordCount: 19&#13;  wordValue: 0&#13;  result: 0&#13;</pre></details></td><td>0.7</td><td>4</td><td>0</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1570010044\" target=\"_blank\" rel=\"noopener\">I think in this case it makes sense because this function will r&hellip;</a></h6></td><td><details><summary>12.49</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 2&#13;    code:&#13;      score: 1&#13;      elementCount: 4&#13;  result: 6&#13;regex:&#13;  wordCount: 60&#13;  wordValue: 0.2&#13;  result: 6.49&#13;</pre></details></td><td>0.7</td><td>4</td><td>84.344</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1570011467\" target=\"_blank\" rel=\"noopener\">It is supposed to represent a comment. Would you prefer a descri&hellip;</a></h6></td><td><details><summary>4.11</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;    code:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 2&#13;regex:&#13;  wordCount: 16&#13;  wordValue: 0.2&#13;  result: 2.11&#13;</pre></details></td><td>0.7</td><td>4</td><td>27.816</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1570014142\" target=\"_blank\" rel=\"noopener\">True, just picked this up from the previous codebase. https://rp&hellip;</a></h6></td><td><details><summary>3.44</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 1&#13;regex:&#13;  wordCount: 19&#13;  wordValue: 0.2&#13;  result: 2.44&#13;</pre></details></td><td>0.7</td><td>4</td><td>21.664</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1570187141\" target=\"_blank\" rel=\"noopener\">Seems convoluted to me and I don't know what problem that solved&hellip;</a></h6></td><td><details><summary>7.6</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 2&#13;    code:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 3&#13;regex:&#13;  wordCount: 40&#13;  wordValue: 0.2&#13;  result: 4.6&#13;</pre></details></td><td>0.7</td><td>4</td><td>49.76</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1570188450\" target=\"_blank\" rel=\"noopener\">Changed &#96;COMMENT&#96; to &#96;COMMENTED&#96;, added jsdoc de&hellip;</a></h6></td><td><details><summary>6.98</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 2&#13;    code:&#13;      score: 1&#13;      elementCount: 2&#13;  result: 4&#13;regex:&#13;  wordCount: 24&#13;  wordValue: 0.2&#13;  result: 2.98&#13;</pre></details></td><td>0.7</td><td>4</td><td>48.688</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1573413974\" target=\"_blank\" rel=\"noopener\">Simply makes the docs linking the code within the comment, for c&hellip;</a></h6></td><td><details><summary>8.37</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 3&#13;  result: 3&#13;regex:&#13;  wordCount: 48&#13;  wordValue: 0.2&#13;  result: 5.37&#13;</pre></details></td><td>0.7</td><td>4</td><td>54.072</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1573416633\" target=\"_blank\" rel=\"noopener\">In the case of a string it would coerce the value and you would &hellip;</a></h6></td><td><details><summary>4.09</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 1&#13;regex:&#13;  wordCount: 25&#13;  wordValue: 0.2&#13;  result: 3.09&#13;</pre></details></td><td>0.7</td><td>4</td><td>25.304</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1573420718\" target=\"_blank\" rel=\"noopener\">&#96;observe&#96; would help to chose what we want to include an&hellip;</a></h6></td><td><details><summary>7.7</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;    code:&#13;      score: 1&#13;      elementCount: 1&#13;    a:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 3&#13;regex:&#13;  wordCount: 41&#13;  wordValue: 0.2&#13;  result: 4.7&#13;</pre></details></td><td>0.7</td><td>4</td><td>50.32</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1573731420\" target=\"_blank\" rel=\"noopener\">Not very useful indeed. I am working on the tests on a separate &hellip;</a></h6></td><td><details><summary>6.31</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 2&#13;  result: 2&#13;regex:&#13;  wordCount: 37&#13;  wordValue: 0.2&#13;  result: 4.31&#13;</pre></details></td><td>0.7</td><td>4</td><td>40.136</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1573731649\" target=\"_blank\" rel=\"noopener\">See https://github.com/ubiquity-os/conversation-rewards/pull/14 &hellip;</a></h6></td><td><details><summary>3.22</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 1&#13;regex:&#13;  wordCount: 17&#13;  wordValue: 0.2&#13;  result: 2.22&#13;</pre></details></td><td>0.7</td><td>4</td><td>20.432</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1573733603\" target=\"_blank\" rel=\"noopener\">Good point, also symbol should not be DAI but ETH isn't it?&#9&hellip;</a></h6></td><td><details><summary>7.89</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 2&#13;    code:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 3&#13;regex:&#13;  wordCount: 43&#13;  wordValue: 0.2&#13;  result: 4.89&#13;</pre></details></td><td>0.7</td><td>4</td><td>51.384</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1573737049\" target=\"_blank\" rel=\"noopener\">Had changed it everywhere else but here, now it should be fine</a></h6></td><td><details><summary>2.65</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 1&#13;regex:&#13;  wordCount: 12&#13;  wordValue: 0.2&#13;  result: 1.65&#13;</pre></details></td><td>0.7</td><td>4</td><td>17.24</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1574114225\" target=\"_blank\" rel=\"noopener\">This matches any whitespace character (spaces, tabs, line breaks&hellip;</a></h6></td><td><details><summary>4.91</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 1&#13;regex:&#13;  wordCount: 33&#13;  wordValue: 0.2&#13;  result: 3.91&#13;</pre></details></td><td>0.7</td><td>4</td><td>29.896</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1574622869\" target=\"_blank\" rel=\"noopener\">By definition &#96;dump&#96; was just outputting results to stdo&hellip;</a></h6></td><td><details><summary>9.22</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;    code:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 2&#13;regex:&#13;  wordCount: 68&#13;  wordValue: 0.2&#13;  result: 7.22&#13;</pre></details></td><td>0.7</td><td>4</td><td>56.432</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1574702577\" target=\"_blank\" rel=\"noopener\">Un-nested them one level. I think that the &#96;buildXrow&#96; s&hellip;</a></h6></td><td><details><summary>5.09</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;    code:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 2&#13;regex:&#13;  wordCount: 25&#13;  wordValue: 0.2&#13;  result: 3.09&#13;</pre></details></td><td>0.7</td><td>4</td><td>33.304</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1574724902\" target=\"_blank\" rel=\"noopener\">I assumed BigInt only handles integers, and we manipulate a lot &hellip;</a></h6></td><td><details><summary>3.98</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 1&#13;regex:&#13;  wordCount: 24&#13;  wordValue: 0.2&#13;  result: 2.98&#13;</pre></details></td><td>0.7</td><td>4</td><td>24.688</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1574767984\" target=\"_blank\" rel=\"noopener\">There is one permit generated per participating user, last run:&hellip;</a></h6></td><td><details><summary>4.44</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 2&#13;  result: 2&#13;regex:&#13;  wordCount: 19&#13;  wordValue: 0.2&#13;  result: 2.44&#13;</pre></details></td><td>0.7</td><td>4</td><td>29.664</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1575659438\" target=\"_blank\" rel=\"noopener\">I made it according to the previous version:- Issue Specificat&hellip;</a></h6></td><td><details><summary>12.8</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 5&#13;    ul:&#13;      score: 1&#13;      elementCount: 1&#13;    li:&#13;      score: 1&#13;      elementCount: 2&#13;  result: 8&#13;regex:&#13;  wordCount: 42&#13;  wordValue: 0.2&#13;  result: 4.8&#13;</pre></details></td><td>0.7</td><td>4</td><td>90.88</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1575675704\" target=\"_blank\" rel=\"noopener\">Wouldn't that lead to precision loss in JS and after the divisio&hellip;</a></h6></td><td><details><summary>5.6</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;    code:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 2&#13;regex:&#13;  wordCount: 30&#13;  wordValue: 0.2&#13;  result: 3.6&#13;</pre></details></td><td>0.7</td><td>4</td><td>36.16</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1576567583\" target=\"_blank\" rel=\"noopener\">That is correct. I understand now why you're confused:The valu&hellip;</a></h6></td><td><details><summary>21.95</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 8&#13;    code:&#13;      score: 1&#13;      elementCount: 7&#13;  result: 15&#13;regex:&#13;  wordCount: 65&#13;  wordValue: 0.2&#13;  result: 6.95&#13;</pre></details></td><td>0.7</td><td>4</td><td>158.92</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1577477616\" target=\"_blank\" rel=\"noopener\">I don't think is it relevant in this scenario because we are not&hellip;</a></h6></td><td><details><summary>6.37</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 1&#13;regex:&#13;  wordCount: 48&#13;  wordValue: 0.2&#13;  result: 5.37&#13;</pre></details></td><td>0.7</td><td>4</td><td>38.072</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#discussion_r1578162024\" target=\"_blank\" rel=\"noopener\">@whilefoo you are correct, these were wrongly used. I corrected &hellip;</a></h6></td><td><details><summary>16.31</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 5&#13;    code:&#13;      score: 1&#13;      elementCount: 5&#13;  result: 10&#13;regex:&#13;  wordCount: 58&#13;  wordValue: 0.2&#13;  result: 6.31&#13;</pre></details></td><td>0.7</td><td>4</td><td>115.336</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#issuecomment-2056635516\" target=\"_blank\" rel=\"noopener\">Example of successful comment posting with results:https://git&hellip;</a></h6></td><td><details><summary>4.11</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 2&#13;  result: 2&#13;regex:&#13;  wordCount: 16&#13;  wordValue: 0.2&#13;  result: 2.11&#13;</pre></details></td><td>0.7</td><td>4</td><td>27.816</td></tr><tr><td><h6><a href=\"https://github.com/ubiquity-os/conversation-rewards/pull/12#issuecomment-2063348339\" target=\"_blank\" rel=\"noopener\">@0x4007 I believe that if I had all the tests in this PR it will&hellip;</a></h6></td><td><details><summary>4.29</summary><pre>content:&#13;  content:&#13;    p:&#13;      score: 1&#13;      elementCount: 1&#13;  result: 1&#13;regex:&#13;  wordCount: 27&#13;  wordValue: 0.2&#13;  result: 3.29&#13;</pre></details></td><td>0.7</td><td>4</td><td>26.424</td></tr></tbody></table></details>

ishowvel commented:
@gentlementlegen are these the desired outputs?

> ###  [ 818.482 WXDAI ] 
> ######
> 			
> 			
> 			
> 			
> 			
> 			
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> 	
> 
> 		
> ###  [[ 2073.928 WXDAI ]](\"undefined\") 
> ######

gentlementlegen commented:
@ishowvel try with priority 3 as it is the most likely to give very long decimals if not properly handled. My comment was about the `0.3000000004` number in my screenshot.

Coming back to my calculation question, here is the detail:
![image](https://github.com/user-attachments/assets/86441067-78c1-4410-90ae-2dfe15da071b)

I can't seem to understand how we get `11.264`.
`2 + 1.88 * 0.8 = 3.504` which would be the result on priority 1, like seen in the screenshot here
![image](https://github.com/user-attachments/assets/56e2ad11-5438-455e-a975-305945ade9c1)

So how does priority 3 make it `11.264`?

0x4007 commented:
I guess 3x multiplier?

0x4007 commented:
I guess 3x multiplier?

gentlementlegen commented:
@0x4007 Sure but `3.504 * 3 = 10.512` so I don't understand where this result comes from.

ishowvel commented:
https://github.com/ubiquity-os-marketplace/text-conversation-rewards/blob/f1e0ab4a3a3537a3bbd1cbed097e2c6cb0bf477d/src/parser/formatting-evaluator-module.ts#L72
cant seem to understand why theres a check for comment.score.reward as this check is bound to return always false, the first time comment.score.reward is given a value is in this very module

ishowvel commented:
heres the new [qa](https://github.com/ishowvel-org/.ubiquity-os/issues/13) @gentlementlegen, the problem was not adding the priority mul at the right time, i should have added at the absolute end

gentlementlegen commented:
> https://github.com/ubiquity-os-marketplace/text-conversation-rewards/blob/f1e0ab4a3a3537a3bbd1cbed097e2c6cb0bf477d/src/parser/formatting-evaluator-module.ts#L72
> 
> 
> cant seem to understand why theres a check for comment.score.reward as this check is bound to return always false, the first time comment.score.reward is given a value is in this very module

Because every module can be disabled individually, the type can technically be undefined (not within this module I agree) so this has to be done for TypeScript not to complain. I'll do a new QA as well.

gentlementlegen commented:
New QA is ok: https://github.com/Meniole/text-conversation-rewards/issues/27#issuecomment-2456148025

Before merging, I would need that you add a `Jest` test with this feature thank you.

ishowvel commented:
Done!

ishowvel commented:
All tests that failed, failed because of not supporting this feature. So I just added support in all of them

ishowvel commented:
@gentlementlegen can you please review these changes? Thank you!

ishowvel commented:
Any updates?

ishowvel commented:
@0x4007 can you review this as you have created the specification, gentlemengen may be busy

gentlementlegen commented:
@ishowvel Sorry I have been extremely busy yes, I'll try to have a look today. In the meantime please fix the conflicts.

ishowvel commented:
Currently out for a university, will work on this task in a few hours

ishowvel commented:
idk why the tests keep failing even without the changes present in this pull request, i made a branch and force reset it to this repo's dev branch, heres the [test](https://github.com/ishowvel/text-conversation-rewards/actions/runs/11758209597/job/32756314494)

gentlementlegen commented:
@ishowvel tests don't pass because the import types don't match, I think `@octokit/rest` is imported when it should be `@octokit/core` within the tests.

ishowvel commented:
@gentlementlegen @octokit/core is imported, this has become a major blocker for me as i cant debug my other [pr](https://github.com/ubiquity-os-marketplace/text-conversation-rewards/pull/182) without working tests

ishowvel commented:
tests even fail on a fresh clone of this repository

gentlementlegen commented:
@ishowvel merge the development branch into this one.

ishowvel commented:
Tests are passing 🎉 🎉 🎉

gentlementlegen commented on src/parser/content-evaluator-module.ts:
```diff
@@ -119,11 +120,14 @@ export class ContentEvaluatorModule extends BaseModule {
-      const currentReward = this._getRewardForComment(currentComment, currentRelevance);
+      const currentReward = this._getRewardForComment(currentComment, currentRelevance)
+        .mul(currentComment.score?.priority ?? 1)
+        .toDecimalPlaces(3);
```
Why cutting to 3 decimals?

ishowvel replied:
The permit rewards have payouts upto 3 decimals, decimal js multiplication doesn't round it up to 3 decimals that's why and I don't get exactly what you mean by precision being lost as 0.1+0.2 always has that weird decimal
https://stackoverflow.com/questions/50778431/why-does-0-1-0-2-return-unpredictable-float-results-in-javascript-while-0-2

gentlementlegen replied:
Yes that is why we use DecimalJs so `0.1 + 0.2` will yield `0.3`. If any rounding happing it should be solely for display or at the end of the computations.

ishowvel replied:
ohhhh my bad

ishowvel commented:
@gentlementlegen can you please review this

ishowvel commented:
Do reminders exist for reviewers like the ones for assignee

gentlementlegen commented:
@ishowvel I tried to run the plugin with your pull-request and this is what I got
https://github.com/Meniole/text-conversation-rewards/actions/runs/11868228402/job/33077194863#step:2:173

Does it work on your side?

ishowvel commented:
From what I have tried using @octokit/rest makes the build work but makes the tests fail and @octokit/core makes the build fail and test work

ishowvel commented:
My code changes do not produce this bug as you can see here
https://github.com/ubiquity-os-marketplace/text-conversation-rewards/actions/runs/11769854878
The latest merge is also failing because of the same reason

gentlementlegen commented:
@ishowvel Okay let me try to fix the test and build then.

gentlementlegen commented:
Seems to be ok in my latest QA: https://github.com/Meniole/text-conversation-rewards/issues/28
